### PR TITLE
RHS Compat - Change IHADSS hearing to crew, add pilot visors

### PR DESCRIPTION
**Pull request by Blutze**
**When merged this pull request will:**
- Change ace_hearing config on the IHADSS helmet to crew helmet level (0.6 muffling, 0.85 protection) instead of Peltor level (no muffling, 0.75 protection). While it inherits from a random FAST helmet, it is actually a heli pilot helmet; RHS's inheritance structure is just that nonsensical.
- Add ace_goggles `ACE_Protection` (protects from post processing caused by rotor downwash) to pilot helmets with visors for RHS blue and red (there don't seem to be any new items in green or other green). Inheritance branches out pretty heavily, so this is about as short as that code is gonna get.

### CODE REVIEW

File: optionals/compat_rhs_afrf3/CfgWeapons.hpp [hunk 1]
```diff
@@ -128,6 +128,17 @@ class CfgWeapons {
     class rhs_zsh7a: H_HelmetB {
         HEARING_PROTECTION_VICCREW
     };
+    class rhs_zsh7a_alt : rhs_zsh7a {
```
Comment (jonpas) [inline optionals/compat_rhs_afrf3/CfgWeapons.hpp]:
```suggestion
    class rhs_zsh7a_alt: rhs_zsh7a {
```

File: optionals/compat_rhs_afrf3/CfgWeapons.hpp [hunk 2]
```diff
@@ -128,6 +128,17 @@ class CfgWeapons {
...
+        ACE_Protection = 1;
+    };
+    class rhs_zsh7a_mike;
+    class rhs_zsh7a_mike_alt : rhs_zsh7a_mike {
```
Comment (jonpas) [inline optionals/compat_rhs_afrf3/CfgWeapons.hpp]:
```suggestion
    class rhs_zsh7a_mike_alt: rhs_zsh7a_mike {
```

File: optionals/compat_rhs_afrf3/CfgWeapons.hpp [hunk 3]
```diff
@@ -128,6 +128,17 @@ class CfgWeapons {
...
+        ACE_Protection = 1;
+    };
+    class rhs_zsh7a_mike_green;
+    class rhs_zsh7a_mike_green_alt : rhs_zsh7a_mike_green {
```
Comment (jonpas) [inline optionals/compat_rhs_afrf3/CfgWeapons.hpp]:
```suggestion
    class rhs_zsh7a_mike_green_alt: rhs_zsh7a_mike_green {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 4]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
         HEARING_PROTECTION_PELTOR
     };
 
-
     class rhsusf_hgu56p: rhsusf_opscore_01 {
         HEARING_PROTECTION_VICCREW
     };
+    class rhsusf_hgu56p_visor : rhsusf_hgu56p {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor: rhsusf_hgu56p {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 5]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_black;
+    class rhsusf_hgu56p_visor_black : rhsusf_hgu56p_black {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_black: rhsusf_hgu56p_black {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 6]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    class rhsusf_hgu56p_usa;
+    class rhsusf_hgu56p_visor_usa : rhsusf_hgu56p_usa {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_white;
+    class rhsusf_hgu56p_visor_white : rhsusf_hgu56p_white {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_black_skull;
+    class rhsusf_hgu56p_mask_black_skull : rhsusf_hgu56p_visor_mask_black_skull {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_mask_black_skull: rhsusf_hgu56p_visor_mask_black_skull {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 7]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    class rhsusf_hgu56p_saf;
+    class rhsusf_hgu56p_visor_saf : rhsusf_hgu56p_saf {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_usa;
+    class rhsusf_hgu56p_visor_usa : rhsusf_hgu56p_usa {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_white;
+    class rhsusf_hgu56p_visor_white : rhsusf_hgu56p_white {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_white: rhsusf_hgu56p_white {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 8]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    };
+    class rhsusf_hgu56p_visor_pink : rhsusf_hgu56p_pink {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_saf;
+    class rhsusf_hgu56p_visor_saf : rhsusf_hgu56p_saf {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_usa;
+    class rhsusf_hgu56p_visor_usa : rhsusf_hgu56p_usa {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_usa: rhsusf_hgu56p_usa {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 9]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    };
+    class rhsusf_hgu56p_pink;
+    class rhsusf_hgu56p_visor_mask_pink : rhsusf_hgu56p_pink {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_pink : rhsusf_hgu56p_pink {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_saf;
+    class rhsusf_hgu56p_visor_saf : rhsusf_hgu56p_saf {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_saf: rhsusf_hgu56p_saf {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 10]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    };
+    class rhsusf_hgu56p_mask_smiley;
+    class rhsusf_hgu56p_visor_mask_smiley : rhsusf_hgu56p_mask_smiley {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_pink;
+    class rhsusf_hgu56p_visor_mask_pink : rhsusf_hgu56p_pink {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_pink : rhsusf_hgu56p_pink {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_pink: rhsusf_hgu56p_pink {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 11]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    };
+    class rhsusf_hgu56p_visor_mask_green : rhsusf_hgu56p_green {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_mask_smiley;
+    class rhsusf_hgu56p_visor_mask_smiley : rhsusf_hgu56p_mask_smiley {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_pink;
+    class rhsusf_hgu56p_visor_mask_pink : rhsusf_hgu56p_pink {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask_pink: rhsusf_hgu56p_pink {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 12]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_Empire_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_green : rhsusf_hgu56p_green {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_mask_smiley;
+    class rhsusf_hgu56p_visor_mask_smiley : rhsusf_hgu56p_mask_smiley {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask_smiley: rhsusf_hgu56p_mask_smiley {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 13]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    class rhsusf_hgu56p_visor_mask : rhsusf_hgu56p {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_Empire_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_green : rhsusf_hgu56p_green {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask_green: rhsusf_hgu56p_green {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 14]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    class rhsusf_hgu56p_visor_green : rhsusf_hgu56p_green {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask : rhsusf_hgu56p {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_Empire_black : rhsusf_hgu56p_black {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask_Empire_black: rhsusf_hgu56p_black {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 15]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_green;
+    class rhsusf_hgu56p_visor_green : rhsusf_hgu56p_green {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask : rhsusf_hgu56p {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask_black : rhsusf_hgu56p_black {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask_black: rhsusf_hgu56p_black {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 16]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
+    };
+    class rhsusf_hgu56p_black;
+    class rhsusf_hgu56p_visor_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_green;
+    class rhsusf_hgu56p_visor_green : rhsusf_hgu56p_green {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_visor_mask : rhsusf_hgu56p {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_mask: rhsusf_hgu56p {
```

File: optionals/compat_rhs_usf3/CfgWeapons.hpp [hunk 17]
```diff
@@ -413,13 +410,66 @@ class CfgWeapons {
...
     };
+    class rhsusf_hgu56p_visor : rhsusf_hgu56p {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_black;
+    class rhsusf_hgu56p_visor_black : rhsusf_hgu56p_black {
+        ACE_Protection = 1;
+    };
+    class rhsusf_hgu56p_green;
+    class rhsusf_hgu56p_visor_green : rhsusf_hgu56p_green {
```
Comment (jonpas) [inline optionals/compat_rhs_usf3/CfgWeapons.hpp]:
```suggestion
    class rhsusf_hgu56p_visor_green: rhsusf_hgu56p_green {
```

**Comment by Blutze** [comment]
Spaces removed.

**Comment by Blutze** [comment]
Tested, nothing explodes, IHADSS shows hearing protection in Arsenal, downwash protection is active on visored helmet and correctly inactive on IHADSS.

**Pull request merged**